core/src/com/unciv/ui/trade/DiplomacyScreen.kt

```diff
@@ -275,6 +275,20 @@ class DiplomacyScreen(val viewingCiv:CivilizationInfo):CameraStageBaseScreen() {
             }
         }
 
+        for (unique in viewingCiv.getMatchingUniques("Can spend Gold to annex or puppet a City-State that has been your ally for [] turns.")) {
+            val diplomaticMarriageButton =
+                "Diplomatic Marriage ([${otherCiv.getDiplomaticMarriageCost()}] Gold)".toTextButton()
+            diplomaticMarriageButton.onClick {
+                val newCities = otherCiv.cities
+                otherCiv.diplomaticMarriage(viewingCiv)
+                UncivGame.Current.setWorldScreen() // The other civ will no longer exist
```

Comment:
This or re-enter DiplomacyScreen - - - or rightSideTable.clear(); updateLeftSideTable() ?

Reply:
matter of taste imo, I like going back to the worldscreen and getting the popup at once.

Reply:
OK yes that's a very good reason...